### DISCUSSION TITLE
How to avoid image download of same image by multiple jobs triggered by the same job

### DISCUSSION THREAD

**Discussion by klhex** [concourse/concourse#7180] [Help & Support] [answered]
Hello there.

I have several jobs within one pipeline which use the same task (file). (The task's shell code is provided the name of the underlying job and does some things differently based on the job name.) The task uses a specific docker image as runtime environment ("image_resource") that lies in a local docker image repository and gets downloaded from there when the task is executed.

Additionally, I have a separate job which can be used to trigger the parallel execution of the above-mentioned jobs. The issue I have with the parallel execution is that each of the above-mentioned job downloads a copy of the bespoken task image (which is pretty big). For the sake of efficiency, I'd like the trigger job to download the bespoken image and "provide" it to the triggered jobs so that they don't need to download it again for themselves.

I thought I'd found a fitting solution in the [Concourse documentation](https://concourse-ci.org/jobs.html#schema.step.task-step.task)
![image](https://user-images.githubusercontent.com/71178731/122349273-63936400-cf4c-11eb-8cf7-6ca04288cce9.png)
but when I tried this out (after changing "registry-image" to "docker-image" to make it work, see below) the triggered jobs (or rather the task in them) were still downloading the image from the local repo.

BTW, IMHO the above example from the Concourse documentation is not correct because "my-task-image" is typed as "registry-image" while being used in the "build-task-image" job along with a "params: { build: ... }" which is not supported by the resource type "registry-image". The image should be typed as "docker-image" instead I think.

Did I get the above example wrong or is there simply a misunderstanding of what "propagating" implies or actually does not imply?

And is there any (other) way to achieve the described re-use of an already downloaded docker image or does this go against the Concourse Way and is not possible?

**Comment by evanchaoli** [comment, marked as answer]
You may consider to upgrade to 7.3.2 and use the feature https://github.com/concourse/concourse/releases/tag/v7.3.0#6660.

**Reply by klhex** [reply]
Many thanks, that sounds exactly what I am looking for! 😄 

Question can be marked as answered from my point of view. 👍 

(Our Concourse OPS team already tried an upgrade to 7.3.2 (from 6.x) but had to revert to 6.x because of [performance issues](https://github.com/concourse/concourse/issues/6589). However, they will retry as soon as the next related fixes are out.)
